search/median_search2.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ *
+ * @details
+ * Given a linked list L[0,....,n] of n numbers, find the middle node.
+ *
+ * Here are some example lists you can use to see how the algorithm works
+ * A = [1,2,3,4,5]
+ * B = [1,2,3,4,5,6]
+ * print median(A) #should be 39
+ * print median(B) #should be 4
+ *
+ * @author [Benjamin Weiss](https://github.com/weiss-ben)
+ */
+
+/**
+ * Definition for singly-linked list.
+ */
+struct ListNode {
+    int val;												// The value stored in the node
+    ListNode *next;											// Pointer to the next node
+    ListNode() : val(0), next(nullptr) {}					// Default constructor
+    ListNode(int x) : val(x), next(nullptr) {}				// Constructor with value for node->val provided
+    ListNode(int x, ListNode *next) : val(x), next(next) {}	// Constructor with values provided for node->val and node->next
+ };
+ 
+ #include <iostream> // for IO operations
+ #include <cassert>  // for assert in tests
+
+/**
+ * @namespace search
+ * @brief Search algorithms
+ */
+namespace search {
+/**
+ * @namespace median_search
+ * @brief Functions for the Median Search algorithm implementation.
+ */
+namespace median_search2 {
+/**
+* This function searches for the median of a linked list.
+* @param head, The head of the linked list.
+* @returns Median node of the linked list.
+*/  
+ListNode* middleNode(ListNode* head) {
+    if(!head)
+        return nullptr;
+        
+    //Fast and slow pointers
+    ListNode* fastptr;
+    ListNode* slowptr = fastptr = head;
+    
+	// fast jumps 2 while slow jumps 1    
+    while(fastptr->next && fastptr->next->next) {
+        slowptr = slowptr->next;
+        fastptr = fastptr->next->next;
+    }
+    return (fastptr->next) ? slowptr->next : slowptr;
+        
+}
+}  // namespace median_search2
+}  // namespace search
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+ */
```

Comment:
There's an extra comment ending here.
```suggestion
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Not maintainable